### PR TITLE
refactor: add default_font, change font api

### PR DESCRIPTION
**Pull request by simbleau**
### Added

- There is now a `default_font` feature that uses the same `FiraMono-subset.ttf` font used in the bevy/default_font feature.

### Changed

- The font API has changed significantly. Please visit `examples/text` for further usage. This is to prepare for additional text features such as linebreak behavior, bounded text, and text justification.
  - `VelloText` has been renamed to `VelloTextSection`.
  - `VelloText.content` has been renamed to `VelloText.value`.
  - There is now a `VelloTextStyle` struct and it is a required field of `VelloText`.
  - `VelloFont` has been removed from `VelloTextBundle` and moved into `VelloTextStyle`.

### CODE REVIEW

File: src/text/FiraMono-subset.ttf
Comment (DJMcNab) [inline src/text/FiraMono-subset.ttf:1]: What's the license for this file?

Comment (DJMcNab) [inline src/text/FiraMono-subset.ttf:1]: If you are using both this crate with `default_font` and `bevy/default_font`, does your binary include both?

I do see that the file is relatively small, so it's probably fine.

**Review by DJMcNab — COMMENTED** [review]
We need to make sure we're handling the licensing properly.

Can we ask the Bevy project to make a crate containing just the font file, so that the usages can be properly shared?

**Comment by DJMcNab** [comment]
I don't agree with @alice-i-cecile's reading of the OFL in https://github.com/bevyengine/bevy/pull/8445. According to their [FAQ](https://openfontlicense.org/how-to-use-ofl-fonts/):

> The only situation in which an OFL font can be distributed without the text of the OFL (either in a separate file or in font metadata), is when a font is embedded in a document or bundled within a program. In the case of metadata included within a font, it is legally sufficient to include only a link to the text of the OFL on https://openfontlicense.org/, but we strongly recommend against this. Most modern font formats include metadata fields that will accept the full OFL text, and full inclusion increases the likelihood that users will understand and properly apply the license.

I can't find the license text within the font file (at least using `fc-query`), and this use case (hosted on GitHub) is not "bundled within a program" (even though it will be once the program is built).
Even if Bevy has decided it's OK for them, if we're distributing this file, I strongly want us to make it clear that it's under the OFL.

So we should either just remove this file and only use the capability from that PR, or add the OFL.

**Comment by simbleau** [comment]
> I don't agree with @alice-i-cecile's reading of the OFL in [bevyengine/bevy#8445](https://github.com/bevyengine/bevy/pull/8445). According to their [FAQ](https://openfontlicense.org/how-to-use-ofl-fonts/):
> 
> > The only situation in which an OFL font can be distributed without the text of the OFL (either in a separate file or in font metadata), is when a font is embedded in a document or bundled within a program. In the case of metadata included within a font, it is legally sufficient to include only a link to the text of the OFL on https://openfontlicense.org/, but we strongly recommend against this. Most modern font formats include metadata fields that will accept the full OFL text, and full inclusion increases the likelihood that users will understand and properly apply the license.
> 
> I can't find the license text within the font file (at least using `fc-query`), and this use case (hosted on GitHub) is not "bundled within a program" (even though it will be once the program is built). Even if Bevy has decided it's OK for them, if we're distributing this file, I strongly want us to make it clear that it's under the OFL.
> 
> So we should either just remove this file and only use the capability from that PR, or add the OFL.

I always planned on switching with the data from the PR, however I am not opposed to adding the OFL until we can do that. I'd gladly accept a PR to add the OFL.

File: src/text/FiraMono-subset.ttf
Comment (simbleau) [reply src/text/FiraMono-subset.ttf:1]: OFL 1.1

**Pull request merged**